core/domain/classifier_services.py

```diff
@@ -242,6 +242,51 @@ def get_classifier_training_job_from_model(classifier_training_job_model):
         classifier_training_job_model.data_schema_version)
 
 
+def _get_classifier_training_job_model_by_id(job_id, strict=True):
+    """Gets a classifier training job by a job_id.
+
+    Args:
+        job_id: str. ID of the classifier training job.
+        strict: bool. Raises Exception if ClassifierTrainingJobModel with
+            id not found and strict is set to True. Logs an error, otherwise.
+    
+    Returns:
+        classifier_training_job: ClassifierTrainingJobModel. NDB instance
+            of classifier training job.
+    
+    Raises:
+        Exception: Entity for class ClassifierTrainingJobModel with id not
+            found.
+    """
+    classifier_training_job_model = (
+        classifier_models.ClassifierTrainingJobModel.get(job_id, strict=strict))
+    if classifier_training_job_model:
+        classifier_training_job_model.classifier_data = (
+            fs_services.read_classifier_data(
+                classifier_training_job_model.exp_id, job_id))
+    return classifier_training_job_model
+
+
+def _get_classifier_training_jobs_model_by_ids(job_ids):
+    """Gets a classifier training job by a job_id.
+
+    Args:
+        job_ids: list(str). IDs of the classifier training job.
+    
+    Returns:
+        classifier_training_jobs: list(ClassifierTrainingJobModel). NDB
+            instances of classifier training jobs.
+    """
+    classifier_training_job_models = (
+        classifier_models.ClassifierTrainingJobModel.get_multi(job_ids))
+    
+    for job_model in classifier_training_job_models:
+        if job_model:
+            job_model.classifier_data = fs_services.read_classifier_data(
```

Comment:
Same note about not assigning attrs on the fly. Ditto elsewhere.

In general when you create the domain object, make sure it has all the necessary data at initialization.

Reply:
Done.